analyzers/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -90,6 +91,12 @@ internal SonarAnalysisContext(AnalysisContext context, IEnumerable<DiagnosticDes
         public bool ShouldAnalyzeGenerated(Compilation c, AnalyzerOptions options) =>
             ShouldAnalyzeGenerated(context, c, options);
 
+        public bool IsUnchanged(SyntaxTree syntaxTree, Compilation compilation, AnalyzerOptions options) =>
+            UnchangedFilesCache.GetValue(compilation, _ => CreateUnchangedFilesHashSet(options)).Contains(syntaxTree.FilePath);
```

Comment:
I'll need to change the contract a little like this, because we will need to call it from different scenarios.
```
        public static bool IsUnchanged(TryGetValueDelegate<ProjectConfigReader> tryGetValue, SyntaxTree tree, Compilation compilation, AnalyzerOptions options) =>
            UnchangedFilesCache.GetValue(compilation, _ => CreateUnchangedFilesHashSet(tryGetValue, options)).Contains(tree.FilePath);
```
Yes, the design is a mess. And my PR follow up PR makes it even more visible. I plan a cleanup afterwards.

It will need change in `CreateUnchagnedFilesHashSet` below too - see other comment.

In the UTs, it will be invoked as `SonarAnalysisContext.IsUnchanged(sut.TryGetValue)`

Reply:
Why: To make it work. We cannot implement the feature otherwise.

Changing visibility isn't any problem. Nobody references us.

I will have to do a bigger refactoring/cleanup after my PR. That should improve the design significantly. Currently, it's a mess.

Reply:
Or I can do the change in my PR. That's not important.